independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/VertxClientInputStream.java

```diff
@@ -0,0 +1,261 @@
+            long expire = System.currentTimeMillis() + timeout;
+            synchronized (VertxBlockingInput.this) {
+                while (input1 == null && !eof && readException == null) {
+                    long rem = expire - System.currentTimeMillis();
+                    if (rem <= 0) {
+                        //everything is broken, if read has timed out we can assume that the underling connection
+                        //is wrecked, so just close it
+                        request.netSocket().close();
+                        IOException throwable = new IOException("Read timed out");
+                        readException = throwable;
+                        throw throwable;
+                    }
+
+                    try {
+                        if (Context.isOnEventLoopThread()) {
+                            throw new BlockingNotAllowedException("Attempting a blocking read on io thread");
+                        }
+                        waiting = true;
+                        VertxBlockingInput.this.wait(rem);
+                    } catch (InterruptedException e) {
+                        throw new InterruptedIOException(e.getMessage());
+                    } finally {
+                        waiting = false;
+                    }
+                }
+                if (readException != null) {
+                    throw new IOException(readException);
+                }
+                Buffer ret = input1;
+                input1 = null;
+                if (inputOverflow != null) {
+                    input1 = inputOverflow.poll();
+                    if (input1 == null) {
+                        request.fetch(1);
+                    }
+                } else if (!eof) {
+                    request.fetch(1);
+                }
+                return ret == null ? null : ret.getByteBuf();
+            }
+        }
+
+        @Override
+        public void handle(Buffer event) {
+            synchronized (VertxBlockingInput.this) {
+                if (input1 == null) {
+                    input1 = event;
+                } else {
+                    if (inputOverflow == null) {
+                        inputOverflow = new ArrayDeque<>();
+                    }
+                    inputOverflow.add(event);
+                }
+                if (waiting) {
+                    VertxBlockingInput.this.notifyAll();
+                }
+            }
+        }
+
+        public int readBytesAvailable() {
+            if (input1 != null) {
+                return input1.getByteBuf().readableBytes();
+            }
+
+            String length = request.getHeader(HttpHeaders.CONTENT_LENGTH);
+
+            if (length == null) {
+                return 0;
+            }
+
+            try {
+                return Integer.parseInt(length);
+            } catch (NumberFormatException e) {
+                Long.parseLong(length); // ignore the value as can only return an int anyway
```

Comment:
Why bother parsing this?

Reply:
To make sure the value is a valid number? I guess :-)